src/embed_tests/CodecGroups.cs

```diff
@@ -0,0 +1,132 @@
+namespace Python.EmbeddingTest
+{
+    using System;
+    using System.Linq;
+    using NUnit.Framework;
+    using Python.Runtime;
+    using Python.Runtime.Codecs;
```

Comment:
While I don't have a particular problem with this, why are you using this style? It's not in use in the established codebase and I have never seen it outside either.

Reply:
Imagine we use a `System.Uri` a lot. At some point in the future we decide to take a dependency (or introduce our own) on something, that has `Python.Uri` class, which we might not need ourselves, but it is public.

With

```csharp
using System;

namespace Python.Runtime {
  class SomeClass {
    Uri uriField;
  }
}
```

the `uriField` will silently change its type from `System.Uri` to `Python.Uri`.

If the `using System;` is inside `namespace Python.Runtime`, that won't happen.